test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -393,8 +393,10 @@ public void log(StatusData data) {
     }
 
     // separate method so that this can be checked again after suite scoped cluster is shut down
-    protected static void checkStaticState() throws Exception {
-        MockPageCacheRecycler.ensureAllPagesAreReleased();
+    protected static void checkStaticState(boolean afterClass) throws Exception {
+        if (afterClass) {
```

Comment:
do we need this in general or can we maybe only do this in specific test cases?

Reply:
I'm not sure I view this as a long term fix. I just wanted to full CI run without failures caused by holding bytes in between tests. We maybe should talk about the best way to deal with this.

Reply:
ok understood. maybe mark it with `//nocommit`